pkg/gen/python.go

```diff
@@ -93,3 +106,182 @@ func PythonClient(
 
 	return nil
 }
+
+// pythonKeywords is a map of reserved keywords used by Python 2 and 3.  We use this to avoid generating unspeakable
+// names in the resulting code.  This map was sourced by merging the following reference material:
+//
+//     * Python 2: https://docs.python.org/2.5/ref/keywords.html
+//     * Python 3: https://docs.python.org/3/reference/lexical_analysis.html#keywords
+//
+var pythonKeywords = map[string]bool{
+	"False":    true,
+	"None":     true,
+	"True":     true,
+	"and":      true,
+	"as":       true,
+	"assert":   true,
+	"async":    true,
+	"await":    true,
+	"break":    true,
+	"class":    true,
+	"continue": true,
+	"def":      true,
+	"del":      true,
+	"elif":     true,
+	"else":     true,
+	"except":   true,
+	"exec":     true,
+	"finally":  true,
+	"for":      true,
+	"from":     true,
+	"global":   true,
+	"if":       true,
+	"import":   true,
+	"in":       true,
+	"is":       true,
+	"lambda":   true,
+	"nonlocal": true,
+	"not":      true,
+	"or":       true,
+	"pass":     true,
+	"print":    true,
+	"raise":    true,
+	"return":   true,
+	"try":      true,
+	"while":    true,
+	"with":     true,
+	"yield":    true,
+}
+
+// pyName turns a variable or function name, normally using camelCase, to an underscore_case name.
+func pyName(name string) string {
```

Comment:
Is this copy-pasted from some other part of our codebase (e.g. `pulumi-terraform`)? Would it make more sense to stick this is a utility package somewhere to ease bugfixes?

Reply:
yep, it totally is.

What I'd really love is a retargetable unified Pulumi code generator, but I don't mind checking in utility functions here and there that can be shared between code generators. Do you know where such a utility package would go? I feel weird putting it in `pulumi/pulumi`, since it's not used in that repo, and I'd feel weird putting it in `pulumi-terraform` too since it would be weird for this repo to depend on it.

Reply:
I think `pulumi/pulumi` is a fine place to put it.

Reply:
Opened https://github.com/pulumi/pulumi/issues/2151 to track.

Reply:
FWIW I actually don't think it's a huge deal if we have this code in two places. We're using it to generate diff-able Python code, so we _should_ notice if there is an improper generation somewhere, and it _should_ be trivial to fix.

That said, whatever you all want. :)